server/src/storage/retention.rs

```diff
@@ -226,6 +231,35 @@ mod action {
                 log::error!("Failed to run delete task {err:?}")
             }
         }
+
+        let store = CONFIG.storage().get_object_store();
+        let res = remove_manifest_from_snapshot(store.clone(), &stream_name, dates).await;
+        if let Err(err) = res {
+            log::error!("Failed to update manifest list in the snapshot {err:?}")
+        }
+
+        if let Ok(Some(first_event_at)) = catalog::get_first_event(store, &stream_name).await {
```

Comment:
Why do we need to do this here?

Reply:
This is here to update the first event timestamp after cleanup because after retention cleanup, first event timestamp would be different.